version: "3.9"

services: 
  mongo:
    image: mongo:6.0
    ports:
      - "27017:27017"
    environment:
      - MONGO_INITDB_ROOT_USERNAME=root
      - MONGO_INITDB_ROOT_PASSWORD=example
    volumes:
      - mongo_data:/data/db 
    networks:
      - pets-network

  mongo-express:
    image: mongo-express 
    ports:
      - "8081:8081"
    environment:
      - ME_CONFIG_MONGODB_ADMINUSERNAME=root
      - ME_CONFIG_MONGODB_ADMINPASSWORD=example
      - ME_CONFIG_MONGODB_SERVER=mongo
    depends_on:
      - mongo
    networks:
      - pets-network

  backend:
    build:
      context: ./backend
      dockerfile: Dockerfile
    ports:
      - "8000:8000"
    env_file:
      - .env
    volumes:
      - ./backend/app:/app/app
      - ./backend/tests:/app/tests
    depends_on:
      - mongo
    networks:
      - pets-network
    restart: unless-stopped

  tests:
    build:
      context: ./backend
      dockerfile: Dockerfile
    command: pytest -v --disable-warnings
    env_file:
      - .env.test
    volumes:
      - ./backend/app:/app/app
      - ./backend/tests:/app/tests
    depends_on:
      - mongo
    networks:
      - pets-network
    restart: "no"

volumes:
  mongo_data: 

networks:
  pets-network:
    driver: bridge 

